cmd/kubeadm/app/phases/addons/dns: TestTranslateStubDomainKubeDNSToCoreDNS/If_Hostname_present:_Omit_Hostname unit test is flakey

<!-- Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!

If the matter is security related, please disclose it privately via https://kubernetes.io/security/
-->


**What happened**:

The following unit test case is flakey

```
--- FAIL: TestTranslateStubDomainKubeDNSToCoreDNS (0.00s)
    --- PASS: TestTranslateStubDomainKubeDNSToCoreDNS/valid_call_with_multiple_IPs (0.00s)
    --- PASS: TestTranslateStubDomainKubeDNSToCoreDNS/empty_call (0.00s)
    --- PASS: TestTranslateStubDomainKubeDNSToCoreDNS/valid_call (0.00s)
    --- FAIL: TestTranslateStubDomainKubeDNSToCoreDNS/If_Hostname_present:_Omit_Hostname (0.00s)
        dns_test.go:367: expected to find "\n    bar.com:53 {\n       errors\n       cache 30\n       loop\n       forward . 1.2.3.4:5300\n    }\n    \n    my.cluster.local:53 {\n       errors\n       cache 30\n       loop\n       forward . 2.3.4.5\n    }" or "\n    my.cluster.local:53 {\n       errors\n       cache 30\n       loop\n       forward . 2.3.4.5\n    }\n    \n    bar.com:53 {\n       errors\n       cache 30\n       loop\n       forward . 1.2.3.4:5300\n    }" in output: "\n    my.cluster.local:53 {\n       errors\n       cache 30\n       loop\n       forward . 2.3.4.5\n    }"
    --- PASS: TestTranslateStubDomainKubeDNSToCoreDNS/All_hostname:_return_empty (0.00s)
    --- PASS: TestTranslateStubDomainKubeDNSToCoreDNS/missing_stubDomains (0.00s)
FAIL
FAIL	k8s.io/kubernetes/cmd/kubeadm/app/phases/addons/dns	0.014s
```

**What you expected to happen**:

Testcase should pass.

**How to reproduce it (as minimally and precisely as possible)**:

Run the testcase for 20 times, you will see failures couple of times.

```
make check WHAT=./cmd/kubeadm/app/phases/addons/dns GOFLAGS="-count=20 -v" KUBE_TEST_ARGS='-run ^TestTranslateStubDomainKubeDNSToCoreDNS$'
```

**Test failure history**:
https://k8s-testgrid.appspot.com/sig-node-ppc64le#unit-tests

--- a/cmd/kubeadm/app/phases/addons/dns/dns.go
+++ b/cmd/kubeadm/app/phases/addons/dns/dns.go
@@ -318,7 +318,7 @@ func translateStubDomainOfKubeDNSToForwardCoreDNS(dataField string, kubeDNSConfi
 				return "", errors.Wrap(err, "invalid format to parse for proxy")
 			}
 			if len(proxyIP) == 0 {
-				break
+				continue
 			}
 
 			pStanza := map[string]interface{}{}

